### ISSUE TITLE
Questions about Tapuino functionality

### ISSUE THREAD

**Issue by lovrenco** [sweetlilmre/tapuino#43]
Hello. 

First of all, thank you for this great little device. I bought mine from Edu Arana, some time ago, but tested it just few days ago.
https://github.com/arananet/Tapuino-Reloaded

I have some questions that others might find interesting as well:
1. Is there a PDF manual which describes all Tapuino functions and button combinations?
2. Is there a way to play T64 files directly from SD card on Tapuino? 
3.  If I use some save state functionality by Replay cart and dump state of C64 on floppy to restore it later, can I manually set position on Tapuino at exact percent where it was before I dumped state? For example 75% of loaded TAP file (FW, RW I suppose)?

Sorry if this was asked before. Thank you.

**Comment by sweetlilmre** [comment]
Hi there,

Thanks for your interest, to briefly answer your questions:

1.  Unfortunately not. If anyone feels like writing something up in Markdown and sending it to me or submitting a pull request, that would be great.
2.  T64 is not supported
3.  Not in this version

The good news is that I have released the next version of the hardware / firmware which supports (3) and have plans to document it better. I'll possibly look into T64. The Atmega / Arduino doesn't have sufficient flash and ram for new features and enhancements so I am moving to the ESP32 to provide this.

Check the project out here: https://github.com/sweetlilmre/TapuinoNext

**Comment by lovrenco** [comment]
@sweetlilmre Are you sure you cannot add support for FW-RW to tap file?
Something like, now there are Play, Record, and you add "Play from", and when user selects that he must using forward and back keys find exact position in percent, and confirm it with "play" or "wait" option?

No need to provide save state functionality on the Tapuino device itself.

**Comment by lovrenco** [comment]
@sweetlilmre OK, I tested this using Tapuino reloaded, EasyFlash3 with Retro Replay cart and Las ninja game. 
When game loaded it showed 50% on tap file. I saved the game/backed up  RAM to floppy. 
Turned on machine and loaded save state from floppy, and pressed Play on tapuino on the same file. Till restore was loaded tapuino came to 10% of tap file, and when game loaded from restore, tapuino stopped oading procedure. 
I played first level and when I finished it, tapuino continued to increase percent. I had to wait till it reached 50% and then I heard second level loading music, few percent later. Second level loaded just fine. 

So basically, there is no extra need for some hacks, id just Forward/Rewind function could be added Tapuino qould have one extra functionality. 
I do not know how hard is to do it, and do we need some special programmer to load new firmware, or it can be done from SD card. This tapuino reloaded version does not have and microUSB slot that I can see.

**Comment by sweetlilmre** [comment]
RAM and flash limits are almost at the max, it might be possible to squeeze this function (and accompanying UI) into the code, but it would be tight. At this point in time I don't have the bandwidth to look into this as I am concentrating on the Next, the code is open though so if someone wants to send in a pull request I'd look at merging it (the functionality can be found in the TapuinoNext source code).
